assets/js/Ioda/components/controlPanel/ControlPanel.js

```diff
@@ -30,138 +31,132 @@ const UNITS = {
   WEEK: "week",
 };
 
-class ControlPanel extends Component {
-  constructor(props) {
-    super(props);
+const ControlPanel = ({from, until, searchbar, onTimeFrameChange, onClose, title}) => {
 
-    this.state = {
-      popoutOpen: false,
-      customDuration: 1,
-      customUnit: UNITS.DAY,
-      range: [secondsToUTC(props.from), secondsToUTC(props.until)],
-    };
-  }
+  const [popoutOpen, setPopoutOpen] = useState(false);
+  const [customDuration, setCustomDuration] = useState(1);
+  const [customUnit, setCustomUnit] = useState(UNITS.DAY);
+  const [range, setRange] = useState([secondsToUTC(from), secondsToUTC(until)]);
 
-  handlePopoutOpen = (val) => {
-    this.setState({ popoutOpen: val });
+  const handlePopoutOpen = (val) => {
```

Comment:
Minor nit: from a code clarity perspective, I'm of the opinion that it's preferable to have functions declared with `function`, and reserve `const` for computed values. Not required to change, just my two-cents (it's highly likely that at some in this code, I was also defining functions with `const`).